sdk/python/tests/components/test_graph_components.py

```diff
@@ -172,5 +172,33 @@ def test_handle_parsing_task_volumes_and_mounts(self):
 
 #TODO: Test task name conversion to Argo-compatible names
 
+
+    def test_handle_parsing_nested_graph_component(self):
+        component_text = '''\
+name: Top-level component
+implementation:
+  graph:
+    tasks:
+      task 1:
+        componentRef:
+          spec:
+            name: Nested component
+            inputs:
+            - {name: Input 1}
+            implementation:
+              container:
+                image: busybox
+                command:
+                - program.py
+                - {inputValue: Input 1}
+        arguments:
+          Input 1: 11
+'''
+        struct = load_yaml(component_text)
+        ComponentSpec.from_struct(struct)
```

Comment:
Don't you need to validate the deserialized states?

Reply:
I feel that it's covered by other tests (e.g. ModelBase tests). Maybe I should still check something. What would you check?

Reply:
I didn't see any place that you verified that the spec was set anywhere. Could you point me to the test you are referring to?